packages/backend/src/open_payments/wallet_address/service.ts

```diff
@@ -180,12 +192,31 @@ async function updateWalletAddress(
       update.deactivatedAt = null
     }
 
-    return await walletAddress
-      .$query(deps.knex)
+    const updatedWalletAddress = await walletAddress
+      .$query(trx)
       .patchAndFetch(update)
       .withGraphFetched('asset')
       .throwIfNotFound()
+
+    // Override all existing additional properties if new ones are provided
+    if (additionalProperties) {
+      await WalletAddressAdditionalProperty.query(trx)
+        .where('walletAddressId', id)
+        .delete()
+    }
+    if (additionalProperties?.length) {
+      await WalletAddressAdditionalProperty.query(trx).insert(
+        additionalProperties.map((prop) => ({
+          walletAddressId: id,
+          ...prop
+        }))
```

Comment:
just in case I think we should "clean" the values passed into this. Would probably best to have this "cleaning" made into a fn and also used during create in L150-159.

Reply:
Let me know if you have anything else in mind, but I made a fn that takes the existing check in create (that key/value arent empty) and added a step to trim them to get rid of whitespace. Does that cover what you had in mind?

https://github.com/interledger/rafiki/pull/2769/commits/7c21e2db99974ded6c501dc324cd0198e61a1cae

Reply:
I guess this suggestion kinda veers into the suggestion to de-duplicate https://github.com/interledger/rafiki/pull/2769#discussion_r1635789637

I guess I still prefer to let the caller handle that and not enforce it but I guess if we do this would be the place. Not sure if that's something you had in mind or not.